File: data/readable_format/fortran/file1850.f90
module utility
    implicit none
    integer :: gatto = 0
    integer, private :: gatto_timido = 0

contains

    subroutine subtract(mino, ris, magg)
        integer, intent(in) :: mino
        integer, intent(out) :: ris
        integer, optional, intent(in) :: magg

        gatto_timido = 24
        if (present(magg)) then
            ris = magg - mino
        else
            ris = -mino
        endif
        gatto = 42
    end subroutine subtract

end module utility

program test_utility
    use utility
    implicit none
    integer :: results, gatto_test

    ! Test with optional `magg`
    call subtract(3, results, magg=10)
    print *, "Test with magg: Result =", results
    if (results == 7) then
        print *, "Passed"
    else
        print *, "Failed"
    endif

    ! Test without optional `magg`
    call subtract(3, results)
    print *, "Test without magg: Result =", results
    if (results == -3) then
        print *, "Passed"
    else
        print *, "Failed"
    endif

    ! Check global variable `gatto`
    gatto_test = gatto
    print *, "Global `gatto` = ", gatto_test
    if (gatto_test == 42) then
        print *, "`gatto` test Passed"
    else
        print *, "`gatto` test Failed"
    endif

    ! Note: `gatto_timido` is private, so its value is indirectly tested through the subroutine's behavior
end program test_utility

File: data/readable_format/cpp/file1850.cpp
#include <iostream>

namespace utility {
    int gatto = 0;

    // Function when magg is provided
    void subtract(const int& mino, int& ris, const int& magg) {
        ris = magg - mino;
        gatto = 42;
    }

    // Overloaded function when magg is not provided
    void subtract(const int& mino, int& ris) {
        ris = -mino;  // Assuming negative to match Fortran behavior
        gatto = 42;
    }
}

void testSubtractWithMagg() {
    int result;
    utility::subtract(3, result, 10); // Using magg
    if (result == 7) {
        std::cout << "Test 1 Passed" << std::endl;
    } else {
        std::cout << "Test 1 Failed" << std::endl;
    }
}

void testSubtractWithoutMagg() {
    int result;
    utility::subtract(3, result); // Not using magg
    if (result == -3) {  // Assuming behavior matches Fortran version
        std::cout << "Test 2 Passed" << std::endl;
    } else {
        std::cout << "Test 2 Failed" << std::endl;
    }
}

void testGlobalGatto() {
    if (utility::gatto == 42) {
        std::cout << "Global gatto Test Passed" << std::endl;
    } else {
        std::cout << "Global gatto Test Failed" << std::endl;
    }
}

int main() {
    testSubtractWithMagg();
    testSubtractWithoutMagg();
    testGlobalGatto();

    return 0;
}